### ISSUE TITLE
Query Tool: Cannot copy result set to clipboard.

### ISSUE THREAD

**Issue by akshay-joshi** [akshay-joshi/TestRMToHub#608]
**This issue has been migrated from Redmine.**

Author Name: **Jan Holst Jensen** (Jan Holst Jensen)
Original Redmine Issue: [1337](https://redmine.postgresql.org/issues/1337)

---
After executing a SQL query, I can't seem to find a way to copy the result set to the clipboard. In pgAdmin3 I would click the result table's upper left corner (which selects the whole result table) and press Ctrl-C.

**Comment by akshay-joshi** [comment]
Author Name: **Dave Page** (@dpage)



---

-  assigned_to_id  was configured    as 318 
-  fixed_version_id  was changed from   3   to 32

**Comment by akshay-joshi** [comment]
Author Name: **Dave Page** (@dpage)



---

-  priority_id  was changed from    Normal   to  High

**Comment by akshay-joshi** [comment]
Author Name: **Adam Brusselback** (Adam Brusselback)


---
I'd also like to add that the ability to select each individual column and do the same is something I used a ton in pgAdmin3.  Same with the value from the cell.

**Comment by akshay-joshi** [comment]
Author Name: **Dave Page** (@dpage)



---

-  priority_id  was changed from    High   to  Urgent

**Comment by akshay-joshi** [comment]
Author Name: **Surinder Kumar** (Surinder Kumar)



---

-  status_id  was changed from    New   to  In Progress 
-  assigned_to_id  was changed from   318   to 320

**Comment by akshay-joshi** [comment]
Author Name: **Surinder Kumar** (Surinder Kumar)


---
Putting it aside for now, as per Dave's comment "it looks like we
have to replace the existing grid with an alternate control."

---

-  status_id  was changed from    In Progress   to  New 
-  priority_id  was changed from    Urgent   to  Normal

**Comment by akshay-joshi** [comment]
Author Name: **Dave Page** (@dpage)



---

-  fixed_version_id  was changed from   32   to 33

**Comment by akshay-joshi** [comment]
Author Name: **Surinder Kumar** (Surinder Kumar)



---

-  assigned_to_id  removed   320

**Comment by akshay-joshi** [comment]
Author Name: **Dave Page** (@dpage)


---
Moving to feature tracker as this is not implemented.

---

-  tracker_id  was changed from    Bug   to  Feature 
-  fixed_version_id  removed   33

**Comment by akshay-joshi** [comment]
Author Name: **Dave Page** (@dpage)


---
Hmm, I guess this must have been resolved with the move to SlickGrid, as copying/pasting results certainly works for me now.

---

-  status_id  was changed from    New   to  Rejected

**Comment by akshay-joshi** [comment]
Author Name: **Michael Pavey** (Michael Pavey)


---
Dave Page wrote:
> Hmm, I guess this must have been resolved with the move to SlickGrid, as copying/pasting results certainly works for me now.

What version are you using and what process are you following to perform the copy? On macOS Sierra with pgAdmin 4 v1.1 I have tried the following two approaches without success:
1. Click a cell in the results grid, then hit Cmd+A then Cmd+C (what I used to do in pgAdmin3, if memory serves)
1. Click the checkbox at the top-left to select all rows and then click the Copy icon above the query field.

**Comment by akshay-joshi** [comment]
Author Name: **Dave Page** (@dpage)


---
Number 2 is what worked for me. I'm using the latest code at the moment - you can find a test build for Mac at https://developer.pgadmin.org/~dpage/

**Comment by akshay-joshi** [comment]
Author Name: **Michael Pavey** (Michael Pavey)


---
What joy! Thanks, I tested it and just as you said, it works. That said, it could be even better. At the moment it is using commas to separate fields, and quotation marks to surround all text fields. Nothing wrong with that _per se_, of course, but it does make copying and pasting into both Excel and Google Spreadsheets more complicated than it could be. Using pgAdmin III, I was able to configure it to use tabs to separate fields (and no text qualifiers, IIRC). That means a single-step paste, rather than having to go through a wizard and (in the case of Google Spreadsheets) some manual cleanup after pasting.

I couldn't find a preference for this anywhere in the test build. Should I file a separate feature suggestion somewhere?
